WordPress/openverse-frontend
# Convert VButton story to mdx and add all variants

## Fixes


Related to #1220 by @dhruvkb

## Description


This PR converts VButton story to mdx. I had to add this PR when I was trying to find which button version to use for 'Clear filters' plain text button.
Other changes:
- ~~I replaced `flex` class on the button with `inline-flex` to make sure that `VButton` width is consistent when it's a link or a button. Before that, the button as VLink would take up the full width.~~ This changes the placement of buttons and breaks the layout, as you can see in the CI visual regression logs: https://github.com/WordPress/openverse-frontend/runs/6206764477?check_suite_focus=true
- I used the args constants (variants, types, sizes) from the `VButton` component to make sure the story is always synced with the component itself. 
- added a story with all button variants to be able to see them at a glance when choosing which variant to use. In there you can also see that two variants are probably the same, so we might want to review the button variants in a different PR.

## Testing Instructions

I wish #1334 was merged before this, so you could just click the link after the CI runs.
Without it, run `pnpm storybook`, and look at VButton story. It should work, have controls to check different buttons

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
